cli/tools/bench.rs

```diff
@@ -0,0 +1,541 @@
+        format!(
+          "{} ns/iter ({}..{} ns/iter) {}",
+          ns_op.to_formatted_string(&Locale::en),
+          min_op.to_formatted_string(&Locale::en),
+          max_op.to_formatted_string(&Locale::en),
+          colors::green("ok")
+        )
+      }
+      BenchResult::Ignored => colors::yellow("ignored").to_string(),
+      BenchResult::Failed(_) => colors::red("FAILED").to_string(),
+    };
+
+    println!("{} {}", status, colors::gray(human_elapsed(elapsed.into())));
+  }
+
+  fn report_summary(&mut self, summary: &BenchSummary, elapsed: &Duration) {
+    if !summary.failures.is_empty() {
+      println!("\nfailures:\n");
+      for (description, error) in &summary.failures {
+        println!("{}", description.name);
+        println!("{}", error);
+        println!();
+      }
+
+      println!("failures:\n");
+      for (description, _) in &summary.failures {
+        println!("\t{}", description.name);
+      }
+    }
+
+    let status = if summary.has_failed() || summary.has_pending() {
+      colors::red("FAILED").to_string()
+    } else {
+      colors::green("ok").to_string()
+    };
+
+    println!(
+      "\nbench result: {}. {} passed; {} failed; {} ignored; {} measured; {} filtered out {}\n",
+      status,
+      summary.passed,
+      summary.failed,
+      summary.ignored,
+      summary.measured,
+      summary.filtered_out,
+      colors::gray(human_elapsed(elapsed.as_millis())),
+    );
+  }
+}
+
+fn create_reporter(echo_output: bool) -> Box<dyn BenchReporter + Send> {
+  Box::new(PrettyBenchReporter::new(echo_output))
+}
+
+fn fetch_specifiers(
+  include: Vec<String>,
+  ignore: Vec<PathBuf>,
+) -> Result<Vec<ModuleSpecifier>, AnyError> {
+  collect_specifiers(include, &ignore, is_supported_bench_path)
+}
+
+/// Type check a collection of module and document specifiers.
+async fn check_specifiers(
+  ps: &ProcState,
+  permissions: Permissions,
+  specifiers: Vec<ModuleSpecifier>,
+  lib: emit::TypeLib,
+) -> Result<(), AnyError> {
+  let module_specifiers = specifiers
+    .into_iter()
+    .map(|s| (s, ModuleKind::Esm))
+    .collect();
+  ps.prepare_module_load(
+    module_specifiers,
+    false,
+    lib,
+    Permissions::allow_all(),
+    permissions,
+    true,
+  )
+  .await?;
+
+  Ok(())
+}
+
+/// Test a single specifier as documentation containing test programs, an executable test module or
+/// both.
+async fn bench_specifier(
+  ps: ProcState,
+  permissions: Permissions,
+  specifier: ModuleSpecifier,
+  channel: Sender<BenchEvent>,
+  options: BenchSpecifierOptions,
+) -> Result<(), AnyError> {
+  let mut worker = create_main_worker(
+    &ps,
+    specifier.clone(),
+    permissions,
+    vec![ops::bench::init(channel.clone())],
+  );
+
+  // Enable op call tracing in core to enable better debugging of op sanitizer
+  // failures.
+  worker
+    .execute_script(&located_script_name!(), "Deno.core.enableOpCallTracing();")
+    .unwrap();
```

Comment:
I think we don't want this on by default?

Reply:
Yep, forgot to remove it.